integration/images/volume-copy-up/Makefile

```diff
@@ -15,13 +15,75 @@
 all: build
 
 PROJ=gcr.io/k8s-cri-containerd
-VERSION=1.0
+VERSION=1.1
 IMAGE=$(PROJ)/volume-copy-up:$(VERSION)
 
-build:
-	docker build -t $(IMAGE) .
+# Architectures supported: amd64, arm, arm64, ppc64le and s390x
+ARCH ?= amd64
+# Operating systems supported: linux, windows
+OS ?= linux
+# OS Version for the Windows images: 1809, 1903, 1909 2004, 20H2
+OSVERSION ?= 1809 1903 1909 2004 20H2
 
-push:
-	gcloud docker -- push $(IMAGE)
+# The output type could either be docker (local), or registry.
+# If it is registry, it will also allow us to push the Windows images.
+OUTPUT_TYPE ?= docker
 
-.PHONY: build push
+ALL_OS = linux windows
+ALL_ARCH.linux = amd64
+ALL_OS_ARCH.linux = $(foreach arch, ${ALL_ARCH.linux}, linux-$(arch))
+ALL_ARCH.windows = amd64
+ALL_OSVERSIONS.windows := 1809 1903 1909 2004 20H2
+ALL_OS_ARCH.windows = $(foreach arch, $(ALL_ARCH.windows), $(foreach osversion, ${ALL_OSVERSIONS.windows}, windows-$(arch)-${osversion}))
+ALL_OS_ARCH = $(foreach os, $(ALL_OS), ${ALL_OS_ARCH.${os}})
+
+BASE.linux := busybox
+BASE.windows := mcr.microsoft.com/windows/nanoserver
+BASE := ${BASE.${OS}}
+
+all-container: all-container-docker
+
+all-push: all-container-registry push-manifest
+
+all-container-docker: $(addprefix sub-container-docker-,$(ALL_OS_ARCH.linux))
+all-container-registry: $(addprefix sub-container-registry-,$(ALL_OS_ARCH))
+
+# split words on hyphen, access by 1-index
+word-hyphen = $(word $2,$(subst -, ,$1))
+sub-container-%:
+	$(MAKE) OUTPUT_TYPE=$(call word-hyphen,$*,1) OS=$(call word-hyphen,$*,2) ARCH=$(call word-hyphen,$*,3) OSVERSION=$(call word-hyphen,$*,4) container
+
+container: .container-${OS}-$(ARCH)
+
+.container-linux-$(ARCH):
+	docker buildx build --pull --output=type=${OUTPUT_TYPE} --platform ${OS}/${ARCH} \
+		-t $(IMAGE)-${OS}-${ARCH} --build-arg BASE=${BASE} .
+
+.container-windows-$(ARCH):
+	docker buildx build --pull --output=type=${OUTPUT_TYPE} --platform ${OS}/${ARCH} \
+		-t $(IMAGE)-${OS}-${ARCH}-${OSVERSION} --build-arg BASE=${BASE}:${OSVERSION} .
+
+push-manifest:
+	docker manifest create --amend $(IMAGE) $(shell echo $(ALL_OS_ARCH) | sed -e "s~[^ ]*~$(IMAGE)\-&~g")
+	set -x; for arch in $(ALL_ARCH.linux); do docker manifest annotate --os linux --arch $${arch} ${IMAGE} ${IMAGE}-linux-$${arch}; done
+	# For Windows images, we also need to include the "os.version" in the manifest list, so the Windows node can pull the proper image it needs.
+	# At the moment, docker manifest annotate doesn't allow us to set the os.version, so we'll have to it ourselves. The manifest list can be found locally as JSONs.
+	# See: https://github.com/moby/moby/issues/41417
```

Comment:
is this still required? https://github.com/docker/cli/pull/2578

Reply:
Done